src/components/App/App.js

```diff
@@ -1,65 +1,320 @@
+import { SearchResultContext } from "../../contexts/SearchResultContext";
+import { CurrentPageContext } from "../../contexts/CurrentPageContext";
+import { SavedNewsContext } from "../../contexts/SavedNewsContext";
+import { KeywordContext } from "../../contexts/KeywordContext";
+import { getSearchResults } from "../../utils/newsAPI";
+import { signIn, signUp, checkToken } from "../../utils/auth";
+import { addSavedNews, removeSavedNews } from "../../utils/api";
 import Main from "../Main/Main";
-import About from "../About/About";
 import Footer from "../Footer/Footer";
 import SavedNews from "../SavedNews/SavedNews";
 import RegisterModal from "../RegisterModal/RegisterModal";
 import LoginModal from "../LoginModal/LoginModal";
 import RegisterCompleteModal from "../RegisterCompleteModal/RegisterCompleteModal";
-import SearchList from "../SearchResults/SearchList";
-import Preloader from "../Preloader/Preloader";
-import NothingFound from "../NothingFound/NothingFound";
-import SavedNewsHeader from "../SavedNewsHeader/SavedNewsHeader";
+import MobileNavigation from "../MobileNavigation/MobileNavigation";
+import MobileFooter from "../MobileFooter/MobileFooter";
 
 function App() {
   const [activeModal, setActiveModal] = useState("");
   const [isLoading, setIsLoading] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [isLoggedInLoading, setIsLoggedInLoading] = useState(true);
+  const [isSearching, setIsSearching] = useState(false);
+  const [isSaved, setIsSaved] = useState(false);
+  const [isMobile, setIsMobile] = useState(window.innerWidth <= 768);
+  const [currentUser, setCurrentUser] = useState({});
+  const [currentPage, setCurrentPage] = useState("/");
+  const [hasSearched, setHasSearched] = useState("");
+  const [savedNews, setSavedNews] = useState([]);
+  const [searchResults, setSearchResults] = useState([]);
+  const [searchError, setSearchError] = useState(false);
+  const [keyword, setKeyword] = useState("");
 
-  const openRegisterModal = () => {
-    setActiveModal("signup");
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setCurrentPage(location.pathname);
+  }, [location.pathname]);
+
+  useEffect(() => {
+    const handleMobile = () => {
+      setIsMobile(window.innerWidth <= 767);
+    };
+
+    window.addEventListener("resize", handleMobile);
+    return () => window.removeEventListener("resize", handleMobile);
+  }, []);
+
+  // const openRegisterModal = () => {
+  //   setActiveModal("signup");
+  // };
+
+  const openConfirmationModal = () => {
+    setActiveModal("confirm");
   };
+
   const openLoginModal = () => {
     setActiveModal("login");
   };
 
+  const openMobileModal = () => {
+    setActiveModal("mobile");
+  };
+
   const handleCloseModal = () => {
     setActiveModal("");
   };
+
+  const handleRegister = ({ email, password, name }) => {
+    setIsLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js
  function handleSubmit(request) {
    // start loading
    setIsLoading(true);
    request()
      // we need to close only in `then`
      .then(handleCloseModal) 
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
      .catch(console.error)
      // and in finally we need to stop loading
      .finally(() => setIsLoading(false));
  }

```


You need to pass the **request** into the call of `handleSubmit` like that:

```js

  // here is an example
  const handleAddItem = (item) => {
    // here we create a function that returns a promise
    const makeRequest = () => {
      // `return` lets us use a promise chain `then, catch, finally`
      return addNewItem(item).then((item) => {
        setClothingItems([item, ...clothingItems]);
      });
    };
    // here we call handleSubmit passing the request
    handleSubmit(makeRequest);
  };

```

And for the 14th sprint:

```js
 // here is an example
 function handleProfileFormSubmit(inputValues) {
  // here we create a function that returns a promise 
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return api.editProfile(inputValues).then(setCurrentUser);
  }
  // here we call handleSubmit passing the request
  handleSubmit(makeRequest);
}


```

So, this way you can remove a lot of code duplicating